### PR TITLE
added colored logger

### PR DESCRIPTION
**Pull request by arnobaer**
Added `utils.get_colored_logger()` that returns a configured logger that prints colored text in a TTY using ANSI escape codes.

### CODE REVIEW

File: ugt_fwtools/synth_1_module.py [hunk 1]
```diff
@@ -7,78 +7,72 @@
...
 def main() -> None:
     """Main routine."""
 
     # Parse command line arguments.
     args = parse_args()
-    
-    if not os.path.isfile(args.path):
-        message = f"\n===> no such file {args.path!r}\n"
-        print_error(message) 
-        raise RuntimeError()
```
Comment (arnobaer) [inline ugt_fwtools/synth_1_module.py:36]: Try to avoid raising exceptions without any message (no context).

File: ugt_fwtools/synth_1_module.py [hunk 2]
```diff
@@ -7,78 +7,72 @@
...
-        message = f"{tty_bold_red}{message}{tty_reset}"
-    print(message)
 
 def parse_args():
     """Parse command line arguments."""
     parser = argparse.ArgumentParser()
-    parser.add_argument("mod_id", help="module number (eg. 1)")
-    parser.add_argument("path", metavar="<path>", type=os.path.abspath, help=f"fw build path to config file")
+    parser.add_argument("module_id", type=int, help="module ID (eg. 1)")
+    parser.add_argument("filename", type=os.path.abspath, help=f"build config file (*.cfg)")
```
Comment (arnobaer) [inline ugt_fwtools/synth_1_module.py:17]: Adjusted arguments to match with the other scripts.

File: ugt_fwtools/utils.py [hunk 3]
```diff
@@ -135,3 +136,70 @@ def username():
 
 def vivado_batch(source: str) -> None:
     subprocess.run(["vivado", "-mode", "batch", "-source", source, "-nojournal", "-nolog"]).check_returncode()
+
+
+def colored(text: str, color: Optional[str] = None, on_color: Optional[str] = None, attrs: Optional[str] = None) -> str:
```
Comment (arnobaer) [inline ugt_fwtools/utils.py:141]: This function mimics [termcolor](https://pypi.org/project/termcolor/) that requires Python >= 3.8 (scripts still require to run on Python 3.6).

File: ugt_fwtools/synth_1_module.py (hunk 1 above)
Comment (herbberg) [reply ugt_fwtools/synth_1_module.py:36]: Done!

**Pull request merged**